app/main/routes.py

```diff
@@ -187,6 +189,32 @@ def record(record_id: uuid.UUID):
     return render_template("record.html", record=file_metadata)
 
 
+@bp.route("/download/<uuid:record_id>")
+@access_token_sign_in_required
+def download_record(record_id: uuid.UUID):
```

Comment:
looks good, just need some tests using moto to mock s3:

- user with body A can download record part of body A (i.e correct response object returned)
- user with body A cannot download record part of body B (i.e 404 is raised)
- superuser can download any random record with some body (i.e correct response object returned)
- when s3.get raises an error, then we log the error and raise 404


might also be worth an e2e test using playwright where a user navigates to `/record/<ID>` they have access to and then clicks download.